src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/LinkCaseTxn.kt

```diff
@@ -29,17 +34,73 @@ class LinkCaseTxnBuilderFactory(
 class LinkCaseTxnBuilderRepository(
   val repository: LinkCaseTxnRepository,
   val jdbcTemplate: JdbcTemplate,
+  val caseStatusRepository: ReferenceCodeRepository<CaseStatus>,
 ) {
-  fun save(linkCaseTxn: LinkCaseTxn): LinkCaseTxn = repository.saveAndFlush(linkCaseTxn)
+  fun get(sourceCase: CourtCase, targetCourtCase: CourtCase, offenderCharge: OffenderCharge): LinkCaseTxn = repository.findByIdOrNull(
+    LinkCaseTxnId(
+      caseId = sourceCase.id,
+      combinedCaseId = targetCourtCase.id,
+      offenderChargeId = offenderCharge.id,
+    ),
+  )!!
+  fun insert(linkCaseTxn: LinkCaseTxn) {
+    jdbcTemplate.update("insert into LINK_CASE_TXNS (CASE_ID, COMBINED_CASE_ID, OFFENDER_CHARGE_ID, EVENT_ID) values (?, ?, ?, ?) ", linkCaseTxn.id.caseId, linkCaseTxn.id.combinedCaseId, linkCaseTxn.id.offenderChargeId, linkCaseTxn.courtEvent.id)
+  }
   fun updateCreateDatetime(linkCaseTxn: LinkCaseTxn, whenCreated: LocalDateTime) {
     jdbcTemplate.update("update LINK_CASE_TXNS set CREATE_DATETIME = ? where LINK_CASE_TXNS.CASE_ID = ? and LINK_CASE_TXNS.COMBINED_CASE_ID = ? and LINK_CASE_TXNS.OFFENDER_CHARGE_ID = ?", whenCreated, linkCaseTxn.id.caseId, linkCaseTxn.id.combinedCaseId, linkCaseTxn.id.offenderChargeId)
   }
+  fun lookupCaseStatus(code: String): CaseStatus = caseStatusRepository.findByIdOrNull(CaseStatus.pk(code))!!
 }
 
 class LinkCaseTxnBuilder(
   private val repository: LinkCaseTxnBuilderRepository,
 ) : LinkCaseTxnDsl {
   private lateinit var linkCaseTxn: LinkCaseTxn
+  fun linkCases(
+    sourceCase: CourtCase,
+    targetCase: CourtCase,
+    courtEvent: CourtEvent,
+    whenCreated: LocalDateTime,
+  ): List<LinkCaseTxn> {
+    val chargesToCopy = sourceCase.offenderCharges.filterNot { it.resultCode1?.dispositionCode == "F" }
```

Comment:
Simulate what NOMIS would do when linking cases
1. Move charges from source to target that are non-final
2. Added charges to target court event
3. Create linkCaseTxn (which just could not get JPA to do it - due to the OffenderCharge changing parent maybe??)